akka-actor/src/main/resources/reference.conf

```diff
@@ -907,4 +907,113 @@ akka {
   }
 
 
+  # CoordinatedShutdown is an extension that will perform registered
+  # tasks in the order that is defined by the phases. It is started
+  # by calling CoordinatedShutdown(system).run(). This can be triggered
+  # by different things, for example:
+  # - JVM shutdown hook will by default run CoordinatedShutdown
+  # - Cluster node will automatically run CoordinatedShutdown when it
+  #   sees itself as Exiting
+  # - A management console or other application specific command can 
+  #   run CoordinatedShutdown
+  coordinated-shutdown {
+    # The timeout that will be used for a phase if not specified with 
+    # 'timeout' in the phase
+    default-phase-timeout = 5 s
+    
+    # Terminate the ActorSystem in the last phase actor-system-terminate.
+    terminate-actor-system = on
+    
+    # Exit the JVM (System.exit(0)) in the last phase actor-system-terminate
+    # if this is set to 'on'. It is done after termination of the 
+    # ActorSystem if terminate-actor-system=on, otherwise it is done 
+    # immediately when the last phase is reached.  
+    exit-jvm = off
+    
+    # Run the coordinated shutdown when the JVM process exits, e.g.
+    # via kill SIGTERM signal (SIGINT ctrl-c doesn't work).
+    run-by-jvm-shutdown-hook = on
+  
+    # CoordinatedShutdown will run the tasks that are added to these
+    # phases. The phases can be ordered as a DAG by defining the 
+    # dependencies between the phases.  
+    # Each phase is defined as a named config section with the
+    # following optional properties:
+    # - timeout=15s: Override the default-phase-timeout for this phase.
+    # - recover=off: If the phase fails the shutdown is aborted
+    #                and depending phases will not be executed.
+    # depends-on=[]: Run the phase after the given phases
+    phases {
```

Comment:
Very nice. I think `service-unbind` can be easily implemented directly in akka-http. `service-requests-done` would need tracking of outstanding requests which is nothing which is done right now but could be added by users that need it. So, yes, I think looks very good as a starting point.